api/services/Reddit.js

```diff
@@ -48,7 +48,16 @@ var makeRequest = function (refreshToken, requestType, url, data, rateLimitRemai
         console.log("Error with parsing: " + body);
         return callback("Error with parsing: " + body);
       }
-      callback(err, bodyJson);
+      var callback_error;
+      if (err) {
+        console.log(err);
+        callback_error = err;
+      } else if (response.statusCode != 200) {
+        console.log('Reddit error: ' + requestType + ' request sent to ' + url + ' returned ' + response.statusCode +
+        ' - ' + response.statusMessage + '.\nResponse body: ' + JSON.stringify(bodyJson) + '\nForm data sent: ' + JSON.stringify(data));
```

Comment:
Since errors are now verbosely logged from Reddit.js, I removed various other error-logging lines to avoid duplicate error messages.